_data-prepper/pipelines/configuration/processors/list-to-map.md

```diff
@@ -0,0 +1,303 @@
+---
+layout: default
+title: List to map processor
+parent: Processors
+grand_parent: Pipelines
+nav_order: 55
+---
+
+# list_to_map
+
+The `list-to-map` processor converts a list of objects from an event where each object contains a `key` field, into a map of target keys.
+
+## Configuration
+
+The following table describes the configuration options to generate target keys to object mappings:
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`key` | Yes | String | The key of the fields to be extracted as keys in the generated mappings
+`source` | Yes | String | The list of objects with `key` fields that be converted to keys based on the generated mappings.
+`target` | No | String | The target for the generated map. When not specified, the generated map will be placed in the root directory.
+`value_key` | No | String | When specified, values given a `value_key` in objects contained in the source list will be extracted and converted into the value specified by this option based on the generated map. When not specified, objects contained in the source list retain their original value when mapped.
+`flatten` | No | Boolean | When `true`, values in the generated map output flatten into single items based on the `flattened_element`. Otherwise, objects mapped to values from the generated map appear as lists.
+`flattened_element` | Conditionally | String | The element to keep, either `first` or `last`, when `flatten` is set to `true`.
+
+## Usage
+
+To use the `list_to_map` processor, create a `.log` file to create a list of source objects with values. The following example creates a file named `logs_json.log`.  Because the `source` parameter reads each line in the `.log` file as an event, the object list appears as one line even though it contains three objects.
```

Comment:
This may help avoid the confusion between the file source and the `source` parameter in the list-to-map processor.

Also, there are 4 objects in the list, rather than 3. 

```suggestion
To use the `list_to_map` processor, create a `.log` file to create a list of source objects with values. The following example creates a file named `logs_json.log`.  Because the `file` source reads each line in the `.log` file as an event, the object list appears as one line even though it contains three objects.
```

Reply:
I'll switch this to "multiple" objects.